srtcore/api.cpp

```diff
@@ -1688,6 +1688,26 @@ int CUDTUnited::groupConnect(CUDTGroup* pg, SRT_SOCKGROUPCONFIG* targets, int ar
 
     vector<SRTSOCKET> broken;
 
+    // Return value rules:
+    // In non-blocking mode:
+    // - return Socket ID, if:
+    //   - you requested only one connection in this call
+    // In blocking mode:
+    // - return Socket ID, if:
+    //   - you requested only one connection in this call
+    //   - you connect a group that was not connected yet
+    // - otherwise return 0
+
+    // Leave the last SID value in retval if you had only one
+    // connection to start. Otherwise override it with 0.
+    if (arraysize > 1)
+        retval = 0;
```

Comment:
Given that `arraysize` is 400 lines above from this line, it would make sense to make it `const`, so that no one mistakenly changes it in between.
```c++
int CUDTUnited::groupConnect(CUDTGroup* pg, SRT_SOCKGROUPCONFIG* targets, const int arraysize
```